tests/integration/characters/test_specifications.py

```diff
@@ -149,3 +149,26 @@ class BobKeyInputRequirer(BaseSchema):
 
     result = BobKeyInputRequirer().load(dict(bobkey=bytes(federated_bob.public_keys(DecryptingPower)).hex()))
     assert isinstance(result['bobkey'], bytes)
+
+
+def test_mixed_up_bytestring_validation(mock_messagekit, mock_treasuremap):
+    """
+    Tests that we get a helpful error message when passing the wrong data
+    """
+
+    class MessageKitsOnly(BaseSchema):
+
+        mkit = fields.UmbralMessageKit()
+
+    with pytest.raises(SpecificationError) as e:
+        MessageKitsOnly().load({'mkit': b64encode(bytes(mock_treasuremap)).decode()})
+    assert "Input data seems to be the bytes for a MockTreasureMap and not a MessageKit" in str(e)
```

Comment:
There's an alternative way to do this using pytest:
```python
with pytest.raises(SpecificationError,
                   match="Input data seems to be the bytes for a MockTreasureMap and not a MessageKit"):
    MessageKitsOnly().load({'mkit': b64encode(bytes(mock_treasuremap)).decode()})
```

(Thanks @svenski123 for showing us this!)